frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/ReSchedulHearing.js

```diff
@@ -2,75 +2,145 @@ import React from 'react';
 import { useHistory } from 'react-router-dom';
 import { useState } from 'react';
 import {
-  InfoCard,
-  Stepper,
+  Dropdown,
+  CardLabel,
+  Modal,
   Button,
-  Timeline,
-  InfoButton,
-  PopUp,
-} from "@egovernments/digit-ui-components";
+  CardText,
+} from '@egovernments/digit-ui-react-components';
+import { useTranslation } from 'react-i18next';
 
-import { Dropdown } from '@egovernments/digit-ui-components';
-import { CardLabel } from '@egovernments/digit-ui-react-components';
+const Heading = (props) => {
+  return <h1 className="heading-m">{props.label}</h1>;
+};
+
+const Close = () => (
+  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="#FFFFFF">
+    <path d="M0 0h24v24H0V0z" fill="none" />
+    <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z" />
+  </svg>
+);
+
+const CloseBtn = (props) => {
+  return (
+    <div onClick={props?.onClick} style={props?.isMobileView ? { padding: 5 } : null}>
+      {props?.isMobileView ? (
+        <CloseSvg />
+      ) : (
+        <div className={"icon-bg-secondary"} style={{ backgroundColor: "#505A5F" }}>
+          {" "}
+          <Close />{" "}
+        </div>
+      )}
+    </div>
+  );
+};
 
-const RescheduleHearing = () => {
+const RescheduleHearing = ({ onSelect, onCancel, onDismiss }) => {
+  const { t } = useTranslation();
   const history = useHistory();
+  const mobileDeviceWidth = 780;
+  const [isMobileView, setIsMobileView] = useState(window.innerWidth <= mobileDeviceWidth);
+  const onResize = () => {
+    if (window.innerWidth <= mobileDeviceWidth) {
+      if (!isMobileView) {
+        setIsMobileView(true);
+      }
+    } else {
+      if (isMobileView) {
+        setIsMobileView(false);
+      }
+    }
+  };
+
+  React.useEffect(() => {
+    window.addEventListener('resize', onResize);
+    return () => {
+      window.removeEventListener('resize', onResize);
+    };
+  }, [isMobileView]);
+
   let a = "FSM-2018-22-3-001";
   let b = "John vs Doe";
   let c = "NIA S";
-  const [showpopup, setShowpopup] = useState(true);
 
   const handleNavigate = (path) => {
-    const contextPath = window?.contextPath || ''; // Adjust as per your context path logic
+    const contextPath = window?.contextPath || ''; 
     history.push(`/${contextPath}${path}`);
   };
 
-
   const children = [
```

Comment:
array works, and I know this isn't your development, but can you please replace this

children should ideally be defined where children variable is used, i.e bottom of the file
basically get rid of the variable children and directly place the elements inside Modal > div at EOF.